### PR TITLE
Small cleanups to JSON IO

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -337,7 +337,7 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 	}
 
 	Json::Value contractsOutput = Json::objectValue;
-	for (string const& contractName: m_compilerStack.contractNames())
+	for (string const& contractName: success ? m_compilerStack.contractNames() : vector<string>())
```
Comment (axic) [inline libsolidity/interface/StandardCompiler.cpp:340]: The problem here is that previously, we aborted on `!success` much earlier. Now `sourceNames()` above is valid if `!success`, but `contractNames()` throws and the Why3 (following this) also operates on `sourceNames()`.

File: libsolidity/interface/Exceptions.cpp [hunk 2]
```diff
@@ -33,22 +33,22 @@ Error::Error(Type _type, SourceLocation const& _location, string const& _descrip
 	switch(m_type)
 	{
 	case Type::DeclarationError:
-		m_typeName = "Declaration Error";
```
Comment (chriseth) [inline libsolidity/interface/Exceptions.cpp:36]: Is this just used for json-io?

**Review by chriseth — APPROVED** [review]

File: libsolidity/interface/Exceptions.cpp (hunk 2 above)
Comment (axic) [reply libsolidity/interface/Exceptions.cpp:36]: This specific error or this method?

Comment (chriseth) [reply libsolidity/interface/Exceptions.cpp:36]: Does this change only affect json-io.

Comment (axic) [reply libsolidity/interface/Exceptions.cpp:36]: No, it is also used by jsonCompiler and the CLI to print the error type.

Comment (chriseth) [reply libsolidity/interface/Exceptions.cpp:36]: But then this changes an external interface?

Comment (axic) [reply libsolidity/interface/Exceptions.cpp:36]: It is used in the error messages, so it does change the interface if someone is parsing for these substrings.

Comment (axic) [reply libsolidity/interface/Exceptions.cpp:36]: I am fine adding another method which returns it without spaces. That is in the required by the standard json spec.

**Pull request merged**